tracer.go

```diff
@@ -266,3 +264,17 @@ func (t Tags) Apply(o *StartSpanOptions) {
 		o.Tags[k] = v
 	}
 }
+
+// Tag may be passed as a StartSpanOption to add a tag to new spans.
+type Tag struct {
```

Comment:
well, the reason I think it belongs here is because you're breaking the API with this change by replacing previous tag values with `xxxEnum` versions, so best to provide the final replacement.

Pretty much it just needs this method

```go
func (t Tag) Set(span Span) {
    span.SetTag(t.Key, t.Value)
}
```